src/main/java/net/fabricmc/loader/impl/metadata/DependencyOverrides.java

```diff
@@ -93,37 +95,77 @@ private DependencyOverrides() {
 		}
 
 		reader.endObject();
-		return dependencyOverrides;
+
+		return ret;
 	}
 
-	private static Map<String, Map<String, ModDependency>> readKeys(JsonReader reader) throws IOException, ParseMetadataException {
+	private static List<Entry> readKeys(JsonReader reader) throws IOException, ParseMetadataException {
 		if (reader.peek() != JsonToken.BEGIN_OBJECT) {
 			throw new ParseMetadataException("Dependency container must be an object!", reader);
 		}
 
-		Map<String, Map<String, ModDependency>> containersMap = new HashMap<>();
+		Map<ModDependency.Kind, Map<Operation, List<ModDependency>>> modOverrides = new EnumMap<>(ModDependency.Kind.class);
 		reader.beginObject();
 
 		while (reader.hasNext()) {
 			String key = reader.nextName();
+			Operation op;
+
+			if (key.startsWith("+")) {
+				op = Operation.ADD;
+				key = key.substring(1);
+			} else if (key.startsWith("-")) {
+				op = Operation.REMOVE;
+				key = key.substring(1);
+			} else {
+				op = Operation.REPLACE;
+			}
```

Comment:
The operation should define its key prefix.

Reply:
fixed